repro.in

```diff
@@ -343,10 +384,13 @@ __END__
 
     mkdir -p "$KEYRINGCACHE"
 
+    # Always lock first. Otherwise we might end up...
+    # - doing the same thing again - if using test/lock/mkdir
+    # - with empty directory in the follow-up lock - if using test/mkdir/lock
+    lock 9 "$KEYRINGCACHE/$keyring_package.lock"
```

Comment:
I don't think there is a reason for us to grab a blocking lock here. We would be blocking when someone is fetching packages for the cache just to figure out we never really needed to initialize the keyring. The time would be better spent getting that shared lock and fetching cached packages.

Reply:
On theory that sounds good although I don't think we can go it atm.

In particular, pacstrap does not support the "--downloadonly" pacman option. Even if it did, I think that pacman validates the signature alongside the download process. Which would fail, if we did not do the gpg/keyring package.

Reply:
I could do some testing, but I suspect the keyring download/gpg setup is extremely quick. So the actual blocking log should not matter.